src/Servers/Kestrel/Core/src/Adapter/Internal/RawStream.cs

```diff
@@ -14,11 +14,20 @@ internal sealed class RawStream : Stream
     {
         private readonly PipeReader _input;
         private readonly PipeWriter _output;
+        private readonly bool _throwOnCancelled;
+        private volatile bool _cancelCalled;
 
-        public RawStream(PipeReader input, PipeWriter output)
+        public RawStream(PipeReader input, PipeWriter output, bool throwOnCancelled = false)
         {
             _input = input;
             _output = output;
+            _throwOnCancelled = throwOnCancelled;
+        }
+
+        public void CancelPendingRead()
+        {
+            _cancelCalled = true;
```

Comment:
Does CancelPendingRead need to be synchronized?

Reply:
Not for our usage.

Reply:
I don't even think _cancelCalled *needs* to be volatile since CancelPendingRead acquires and releases the Pipe's `_sync` lock right afterwards.